runtime/executor/src/main/java/edu/snu/coral/runtime/executor/data/block/FileBlock.java

```diff
@@ -120,9 +120,12 @@ private void writeSerializedPartitions(final Iterable<SerializedPartition<K>> se
         final K key = partitionMetadata.getKey();
         if (keyRange.includes(key)) {
           // The key value of this partition is in the range.
+          final byte[] serializedData = new byte[partitionMetadata.getPartitionSize()];
+          fileStream.read(serializedData);
+          final ByteArrayInputStream byteInputStream = new ByteArrayInputStream(serializedData);
```

Comment:
These `reading byte array - making it another stream - reading the stream again` process can lead memory & computation overhead. How about adding a new interface in `DataUtil#deserializePartition` which consumes byte array instead of stream?

Reply:
@sanha `DataUtil#deserializePartition` uses `InputStreamIterator` to decode data, which takes `InputStream` as a parameter. Even if we do not use `InputStreamIterator`, we must call `buildInputStream` to get original data from the source, which means we eventually have to build a stream which contains serialized data. I want some advice for an idea to make `deserializePartition` method to not use stream but only byte array.

Reply:
@sanha Now working with all kind of compression we support. Also not using extra stream and memory.